src/common/mod.rs

```diff
@@ -35,6 +36,28 @@ pub enum CrustUser {
     Client,
 }
 
+/// Corresponds to `CrustUser` roles and additionally include public endpoints to test for
+/// external reachability.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum BootstrapperRole {
+    /// `Node` peers are usually requested to be externally reachable, hence include their
+    /// public endpoints.
+    Node(HashSet<SocketAddr>),
+    /// `Client` peers don't include any addresses, because they are never tested for external
+    /// reachability.
+    Client,
+}
+
+impl BootstrapperRole {
+    /// Converts into `CrustUser`.
+    pub fn as_crust_role(&self) -> CrustUser {
```

Comment:
Insn't `Into` (here) and `From` (for `CrustUser` for symmetry) more natural for these sort of things ?